network_simulator/statistics.py

```diff
@@ -4,6 +4,186 @@
 """
 
 
+from .host import *
+from .link import *
+from .flow import *
+
+
+class LinkStats(object):
+    """
+    Contains information of a link regarding to per-link buffer occupancy,
+    packet loss, and flow rate.
+
+    :ivar list buffer_occupancy: a list of (timestamp, buffer_occupancy)
+    tuples.
+    :ivar list packet_loss_times: a list of timestamps, each corresponding
+    to a packet loss.
+    :ivar list packet_transmit_times: a list of timestamps corresponding
+    to when a packet was transmitted.
+    """
+    def __init__(self, buffer_occupancy=0, packet_loss_times=[],
```

Comment:
buffer_occupancy is a list of (timestamp, buffer_occupancy) tuples, so it should default to [].